### PR TITLE
Referral program implementation

### PR DESCRIPTION
**Pull request by maslenitsa93**
```
./cli_wallet --server-rpc-endpoint=ws://127.0.0.1:8091
set_password qwer
unlock qwer
import_key 5JVFFWRLwz6JoP9kguuRFfytToGU6cLgBVTL9t6NB3D3BQLbUBS
create_account cyberfounder test "{}" "300.000 GOLOS" true
transfer_to_vesting cyberfounder test "30.000 GOLOS" true
transfer cyberfounder test "0.200 GOLOS" "" true
create_account_referral test "0.200 GOLOS" "0.000001 GESTS" testref "{}" {"referrer": "test", "interest_rate": 900, "end_date": "2018-09-26T14:00:00", "break_fee": "0.000 GOLOS"} true
```

### CODE REVIEW

File: libraries/protocol/include/golos/protocol/steem_operations.hpp [hunk 1]
```diff
@@ -25,6 +25,21 @@ namespace golos { namespace protocol {
...
+        struct account_referral_options {
+            account_name_type referrer;
+            uint16_t interest_rate;
+            time_point_sec end_date;
+            asset break_fee;
+
+            void validate() const;
+        };
+
+        typedef static_variant <
```
Comment (afalaleev) [inline libraries/protocol/include/golos/protocol/steem_operations.hpp]: using

File: libraries/protocol/include/golos/protocol/steem_operations.hpp [hunk 2]
```diff
@@ -25,6 +25,21 @@ namespace golos { namespace protocol {
...
+            account_referral_options
+        > account_create_with_delegation_extension;
+
+        typedef flat_set<account_create_with_delegation_extension> account_create_with_delegation_extensions_type;
```
Comment (afalaleev) [inline libraries/protocol/include/golos/protocol/steem_operations.hpp]: using

File: libraries/chain/steem_evaluator.cpp [hunk 3]
```diff
@@ -304,6 +335,10 @@ namespace golos { namespace chain {
             if (o.fee.amount > 0) {
                 _db.create_vesting(new_account, o.fee);
             }
+
+            for (auto& e : o.extensions) {
```
Comment (afalaleev) [inline libraries/chain/steem_evaluator.cpp]: Add checking on HF

**Comment by maslenitsa93** [comment]
Continued here: https://github.com/GolosChain/golos/pull/942

**Pull request closed without merging**